### ISSUE TITLE
Feature request: Support isomorphic layouts

### ISSUE THREAD

**Issue by oemsysadm** [PolyMeilex/Neothesia#170]
Learning a midi keyboard with an isomorphic layout, like the Linnstrument, Lumatone, or a number of grid controllers supporting various isomorphic layouts, is difficult and there are not really any free and open source applications out there which will help you learn to play one.

Neothesia looks like one of the better free "midi-hero" style teaching apps, and it would be great if it had support for these odd-ball layouts for those of us who want to learn to play these instruments.

Since the "falling notes" layout only really works with a keyboard that has a single row of keys, maybe a way to support these types of layouts could be a key filling up in color and the color fading in, with a visual indicator of when it should be played.

Also most of these types of controllers support various layouts on the same controller, so being able to create a custom layout within Neothesia to correspond to your physical layout would be necessary.

**Comment by PolyMeilex** [comment]
> Since the "falling notes" layout only really works with a keyboard that has a single row of keys, maybe a way to support these types of layouts could be a key filling up in color

@Glitchy-Tozier Are you sold on this idea? Or do you have anything better in mind? To me it sounds kinda hard to follow, but I don't have anything better in mind so far.

**Comment by Glitchy-Tozier** [comment]
I don't think multiple rows would present an issue in conjunction with the falling blocks. Personally, It wouldn't bother me at all. The way it works in my mind is that the blocks never actually "hit the keys", but they hit/pass the top of the keyboard. This can still happen with keyboards such as the Jankó keyboard.

I'm hesitant to suggest any major changes such as flashing blocks. If it really bothers you that the block's don't touch the physical keys, I'd suggest this:

For keyboards that are basically re-skins of the traditional piano (left = low notes, right = high notes), you could just put the top section of a traditional piano at the top of the keyboard.

![ResizedImage_2025-08-17_23-41-31_1175_1.jpg](https://github.com/user-attachments/assets/7b21f1c5-c55f-4102-9296-67c7702c330d)

- Keep in mind: I don't think this is necessary. If you also think the blocks just need to hit a line at the top of the keyboard, then it's enough to just visualize the keyboard.
- If you want to get fancy, you could add an option to color that line which the notes need to hit.

Keyboards that aren't piano re-skins probably require a different mode altogether. Not sure how to approach these, but @oemsysadm seems to have some ideas. I'm mainly here to advocate for the janko keyboard. If you've got any questions on how it works, hit me up!
